content/documentation/shortcodes.md

```diff
@@ -0,0 +1,213 @@
+---
+author: dpordomingo
+date: 2017-10-31
+title: "Enforcing a maintainable layout using shortcodes"
+image: /post/shortcodes/header.png
+description: "Use reusable shortcodes instead of custom HTML code to ensure a maintainable layout"
+categories: ["documentation"]
+---
+
+Since we're trying to have a maintainable blog, with a robust layout that should be easily redesigned, we decided to forbid HTML code inside the blog posts.
+
+If you want/need special formatting for your content, you can use [shortcodes](https://gohugo.io/content-management/shortcodes) to use our predefined layouts.
+
+The shortcodes are simple snippets &ndash;that can be placed inside the content files&ndash; that Hugo will render using a predefined template. Hugo comes with some builtin shortcodes, but not all of them are supported by our blog. The full list of available shortcodes is below this introduction.
+
+You can combine most of the shortcodes to mix its effects.
+
+If you feel you really need a new layout (because none of the existent ones, or none of its combinations suits you), please ask for it opening a [new issue](https://github.com/src-d/blog/issues) to analize that special case.
+
+# source{d} custom shortcodes
+
+## Scroll-panel
+
+If your content is too long to be shown all at once, you can use the `scroll-panel` shortcode to create an scrollable container with fixed height.
+
+``````
+{{%/* scroll-panel height="120" %}}
+```js
+// many lines code goes here
+```
+{{% /scroll-panel */%}}
+``````
+
+The minimum height for an `scroll-panel` containing a block of source code should be 120px to obtain a proper visualization.
+
+_example_:
+{{% scroll-panel height="120" %}}
+```js
+console.log("code1")
+console.log("code2")
+console.log("code3")
+console.log("code4")
+console.log("code5")
+console.log("code6")
+console.log("code7")
+```
+{{% /scroll-panel %}}
+## Center
+
+If you want to center a content you should use the `center` shortcode.
+
+```
+{{%/* center %}}
+This content will be centerd
+{{% /center */%}}
+```
+
+_example_:
+{{% center %}}
+This content will be centerd
+{{% /center %}}
+
+## Caption
+
+Whenever you need to use an image in your post, you should consider the `caption` shortcode to provide valuable information to the user; think in the screen readers.
+
+For the caption of the image, it can be used html, markdown and other shortcodes
+
+```
+{{%/* caption src="//url-to-the-omage" title="title of image" %}}
```

Comment:
why does it start with `/*`?

Reply:
since the documentation has been written in a post entry (to use the hugo engine to generate the real display for the examples), the only way I found to avoid hugo to parse the shortcodes source code was introducing them inside comments:

```
{{%/* caption %}}
This shortcode is not parsed as a shortcode
{{% /caption */%}}
```

That way it can be properly rendered:
![image](https://user-images.githubusercontent.com/2437584/32246306-d98dbcf6-be7e-11e7-9db5-66e28ac21181.png)